java-checks/src/main/java/org/sonar/java/symexecengine/DataFlowVisitor.java

```diff
@@ -43,25 +46,46 @@
 import org.sonar.plugins.java.api.tree.WhileStatementTree;
 
 import javax.annotation.CheckForNull;
+
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
 
-public abstract class DataFlowVisitor extends BaseTreeVisitor {
+public class DataFlowVisitor extends BaseTreeVisitor {
 
-  public DataFlowVisitor() {
-    executionState = new ExecutionState();
+  private final SymbolicExecutionCheck check;
+
+  public static void analyze(MethodTree analyzedMethod, SymbolicExecutionCheck check) {
+    new DataFlowVisitor(analyzedMethod, check);
```

Comment:
:new: ![CRITICAL](https://raw.githubusercontent.com/henryju/image-hosting/master/critical.png)Either remove this useless object instantiation of class "DataFlowVisitor" or use it[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1848)